mantidproject/mantid
# Crash for help docs with parent widget

**Description of work.**
This PR fixes an issue where interface documentation would crash on the second time of opening if they had a parent widget. It was caused by a dangling pointer to a `QtHelpUrlHandler` within the `QWebEngineProfile` object in `pqHelpWindow`. This is easily fixed by giving `QtHelpUrlHandler` a parent which is destructed at the appropriate moment. 

There was also a hidden issue where the pointer to `g_helpWindow` was not being cleaned up properly, which would cause a crash. This has been fixed by setting `g_helpWindow = nullptr` in the `MantidHelpWindow` destructor.

Another change made in this PR is the ability to pass a parent widget directly through to the `MantidHelpWindow` constructor via the `AbstractInstantiator` class. This reduces some complexity in the code, which caused difficulty when debugging.

**To test:**
1. Open `General`->`Fit Script Generator` and open the documentation `?`
2. Close the FSG interface and the documentation page should also close
3. Open the FSG again and open the documentation page. There should be no crash. Close this interface again
4. Now repeat this process for another interface, such as `Indirect`->`Bayes`. Closing the Bayes interface **should not** close its documentation page because it isn't a parent (although the documentation page might hide behind the mantid window).
5. Mess around with opening interface documentation and closing in different orders and make sure no crashes or errors
6. Finally, close mantid, and make sure there is no crash or error

Fixes #34116

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
